contrib/test.sh

```diff
@@ -80,15 +88,12 @@ then
 fi
 
 # Bench if told to, only works with non-stable toolchain (nightly, beta).
-if [ "$DO_BENCH" = true ]
-then
-    if [ "NIGHTLY" = false ]
-    then
-        if [ -n "TOOLCHAIN" ]
-        then
-            echo "TOOLCHAIN is set to a non-nightly toolchain but DO_BENCH requires a nightly toolchain"
+if [ "$DO_BENCH" = true ]; then
+    if [ "$STABLE" = true ]; then
+        if [ -n "$RUSTUP_TOOLCHAIN" ]; then
+            echo "RUSTUP_TOOLCHAIN is set to a stable toolchain but DO_BENCH requires a non-stable (beta, nightly) toolchain"
         else
-            echo "DO_BENCH requires a nightly toolchain"
+            echo "DO_BENCH requires a nightly a non-stable (beta, nightly) toolchain"
```

Comment:
sloppy work, this needs fixing. Will wait for any other review before doing it.

Reply:
I don't think it's so bad. Bash is not a super expressive language (or at least, its advanced features are completely opaque :))

Reply:
Oh I just meat the English is bad.